tests/syscall/mount/mount-immutables-chroot.bats

```diff
@@ -0,0 +1,578 @@
+#!/usr/bin/env bats
+
+#
+# Verify trapping & emulation on "mount" and "unmount2" syscalls
+#
+
+load ../../helpers/run
+load ../../helpers/syscall
+load ../../helpers/docker
+load ../../helpers/environment
+load ../../helpers/mounts
+load ../../helpers/sysbox-health
+
+skipTest=0
+
+function teardown() {
+  sysbox_log_check
+}
+
+# Test to verify immutable mounts within a sys container's chroot() context.
+#
+# Note: a sys container immutable mount is a mount that is setup at container
+# creation time.
+#
+
+chrootpath="/tmp/chrootdir"
+immutable_ro_dir="ro_dir"
+immutable_ro_file="ro_file"
+immutable_masked_dir="masked_dir"
+immutable_masked_file="masked_file"
+
+immutable_ro_dir_path="${chrootpath}/${immutable_ro_dir}"
+immutable_ro_file_path="${chrootpath}/${immutable_ro_file}"
+immutable_masked_dir_path="${chrootpath}/${immutable_masked_dir}"
+immutable_masked_file_path="${chrootpath}/${immutable_masked_file}"
+
+function local_rootfs_prepare() {
+  mkdir -p ${immutable_ro_dir_path}
+  touch ${immutable_ro_file_path}
+  mkdir -p ${immutable_masked_dir_path}
+  touch ${immutable_masked_file_path}
+}
+
+# Testcase #1.
+#
+# Ensure immutable mounts can't be unmounted from inside the container
```

Comment:
This test is not operating on the immutable mounts, it's operating on bind mounts from the immutable mounts. To operate on the immutable mount directly, you have to chroot to a parent dir of the immutable mount (e.g., chroot to /usr/src/ and unmount linux-headers-...).

I suspect the same applies to many of the tests below.